packages/lodestar/src/chain/eventHandlers.ts

```diff
@@ -336,8 +338,9 @@ export async function onErrorAttestation(this: BeaconChain, err: AttestationErro
       this.pendingAttestations.putByBlock(err.type.beaconBlockRoot, err.job);
       break;
 
-    default:
-      await this.db.attestation.remove(err.job.attestation);
+    // TODO: Why is necessary to remove the attestation from the DB on error?
```

Comment:
I think this was failsafe since we didn't have api validation before so on some occasion, validator would put some invalid attestation which would prevent all blocks from being processed

Reply:
How could an invalid attestation prevent all blocks from being processed?

Reply:
process block failed when processing attestations and attestation was put in every block we were producing

Reply:
But how could an invalid attestation got included in the attestation pool?

Reply:
this error handler for attestation was for forkchoice, not when we process block.
Since it was for forkchoice, it's not really necessary to remove that attestation from db. The next time we see that same error from that same attestation, we could just throw error and handle the same way but this is rare.